bin/test/main.ml

```diff
@@ -294,62 +294,63 @@ let run_exn (`Setup ()) (`Non_deterministic non_deterministic)
     let active =
       active t && Block.version_enabled t && (not (Block.skip t))
     in
-    match active, non_deterministic, Block.mode t, Block.value t with
+    let print_block () = Block.pp ?syntax ppf t in
+    match active, 
+          non_deterministic, 
+          Block.file t, 
+          Block.mode t, 
+          Block.value t 
+    with
     (* Print errors *)
-    | _, _, _, Error _ -> Block.pp ?syntax ppf t
-    (* Skip or copy raw blocks. Without parts support *)
-    | true, _, _, Raw -> 
-    (match Block.part t with
-      | None -> 
-        (match Block.file t with
-          | Some file -> 
-            let new_content = (read_part file (Block.part t)) in
-            update_block_content ppf t new_content
-          | None -> Block.pp ?syntax ppf t )
-      | Some _ -> Fmt.failwith "Parts are not supported for non-OCaml code blocks.")
+    | _, _, _, _, Error _ 
     (* The command is not active, skip it. *)
-    | false, _, _, _ -> Block.pp ?syntax ppf t
+    | false, _, _, _, _ 
     (* the command is active but non-deterministic so skip everything *)
-    | true, false, `Non_det `Command, _ -> Block.pp ?syntax ppf t
+    | true, false, _, `Non_det `Command, _ -> print_block ()
     (* the command is active but it's output is
        non-deterministic; run it but keep the old output. *)
-    | true, false, `Non_det `Output, Cram { tests; _ } ->
-      Block.pp ?syntax ppf t;
+    | true, false, _, `Non_det `Output, Cram { tests; _ } ->
+      print_block ();
       let blacklist = Block.unset_variables t in
       List.iter (fun t ->
           let _: int = run_test ?root blacklist temp_file t in ()
         ) tests
-    | true, false, `Non_det `Output, Toplevel tests ->
+    | true, false, _, `Non_det `Output, Toplevel tests ->
       assert (syntax <> Some Cram);
-      Block.pp ppf t;
+      print_block ();
       List.iter (fun test ->
           match eval_test t ?root c test with
           | Ok _    -> ()
           | Error e ->
             let output = List.map (fun l -> `Output l) e in
             if Output.equal test.output output then ()
             else err_eval ~cmd:test.command e
-        ) tests
-    (* Run raw OCaml code *)
-    | true, _, _, OCaml ->
-      assert (syntax <> Some Cram);
-      (match Block.file t with
-       | Some ml_file ->
-         update_file_or_block ?root ppf file ml_file t direction
-       | None ->
-         eval_raw t ?root c ~line:t.line t.contents;
-         Block.pp ppf t )
-    (* Cram tests. *)
-    | true, _, _, Cram { tests; pad } ->
-      run_cram_tests ?syntax t ?root ppf temp_file pad tests
-    (* Top-level tests. *)
-    | true, _, _, Toplevel tests ->
-      assert (syntax <> Some Cram);
-      match Block.file t with
-      | Some ml_file ->
-        update_file_or_block ?root ppf file ml_file t direction
-      | None ->
+        ) tests     
+
```

Comment:
I'm not a huge fan of empty lines in the middle of functions. Could you please remove it?